packages/apps/graph/src/graph/objects/block.ts

```diff
@@ -131,6 +129,47 @@ export default builder.prismaNode('Block', {
       },
     }),
 
+    events: t.prismaConnection({
+      type: 'Event',
+      cursor: 'blockHash_orderIndex_requestKey',
+      edgesNullable: false,
+      complexity: (args) => ({
+        field: getDefaultConnectionComplexity({
+          first: args.first,
+          last: args.last,
+        }),
+      }),
+      select: {
+        hash: true,
+      },
+      async totalCount(parent) {
+        try {
+          return await prismaClient.event.count({
+            where: {
+              blockHash: (parent as Block).hash,
+            },
+          });
+        } catch (error) {
+          throw normalizeError(error);
+        }
+      },
+      async resolve(query, parent) {
+        try {
+          return await prismaClient.event.findMany({
+            ...query,
+            where: {
+              blockHash: (parent as Block).hash,
+            },
+            orderBy: {
+              orderIndex: 'asc',
+            },
+          });
+        } catch (error) {
+          throw normalizeError(error);
+        }
+      },
```

Comment:
I was thinking. If we on the query level select events, then here we wouldn't need to make another query. We could just get the events from the parent object. 
So on `query.object` I'd look like this:

```ts
  try {
    const block = await prismaClient.block.findUnique({
      ...query,
      where: {
        hash: args.hash,
      },
      select: {
        ...query.select,
        events: true,
      },
    });
```

I am not sure if this is the best approach, let me know your thoughts :)

Reply:
Forget it, found a better approach. See https://github.com/kadena-community/kadena.js/pull/1690/files#r1497421049